mm/routines/product.py

```diff
@@ -15,3 +15,13 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
                 sku, att_name, att_value)
 
         return record
+
+    def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
+        rows = \
+            self.call(
+                'get_configurable_associated_products',
+                store_id,
+                int(is_active) if isinstance(is_active, bool) else None,
```

Comment:
I would never allow garbage to get through. If there's garbage, then
someone is asleep at the wheel.

If I want to make assumptions to make things easier to read, I'll always
drop assertions at the top. In this case, I'd just assert that "is_active
is True or is_active is False or is_active is None". Then, you can take out
that inelegant inline if/then.

Of course, this is Pythonic according to what is easiest to read and
understand. There aren't any particular patterns/axioms that come to mind.
People don't like the inline if/then's but people don't also seem to use a
lot of assertions. I guess I'd just chalk it up to people not being willing
to be as strict as I am.

On Wed, May 18, 2016 at 11:40 PM, Diego Terzano <notifications@github.com>
wrote:

> In mm/routines/product.py
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4#discussion_r63818901>
> :
>
> > @@ -15,3 +15,13 @@ def update_enum_product_attribute(self, sku, att_name, att_value):
> >                  sku, att_name, att_value)
> >
> >          return record
> > +
> > +    def get_configurable_associated_products(self, store_id=None, is_active=None, is_visible=None):
> > +        rows = \
> > +            self.call(
> > +                'get_configurable_associated_products',
> > +                store_id,
> > +                int(is_active) if isinstance(is_active, bool) else None,
>
> if they pass garbage, I want the default to apply (None) while allowing
> None/True/False as valid values.
> How would you express it? Let me know if there is a more Pythonic way.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/CoffeeForThinkers/MagentoModels/pull/4/files/9aa86cb203d8cf0a259523bfe5481f93b0514543#r63818901>
>